tests/networking-hosts.nix

```diff
@@ -0,0 +1,16 @@
+{ config, pkgs, ... }:
+
+{
+  networking.hosts = {
+    "127.0.0.1" = [ "my.super.host" ];
+    "10.0.0.1" = [ "my.super.host" "my.other.host" ];
+  };
+
+  test = ''
+    echo checking /etc/hosts file >&2
+
+    file=${config.out}/etc/hosts
+    grep '127.0.0.1' $file
```

Comment:
Probably also worth testing that `localhost` is mapped to `127.0.0.1` and `::1` as they're the defaults